plugins/gpio/GPIODriver.cpp

```diff
@@ -48,6 +48,7 @@ using std::vector;
 
 GPIODriver::GPIODriver(const Options &options)
     : m_options(options),
+      m_first_slot(options.start_address - 1),
```

Comment:
Can we stick to calculating this each time UpdateGPIOPins is called please. I know it's not 100% optimised, but it means when we finally get stuff like dynamic config this won't need fixing and it just generally reduces the risk of weird edge cases. Other plugins like SPI do it as follows:
`const unsigned int first_slot = m_options.start_address - 1;`

Reply:
Makes sense.  I'm traveling this week so may not be able to get to it until the weekend.

Reply:
That's fine, it may take me a few days after that to review it too.

Reply:
@peternewman pushed the change.  It will work when m_options is updated dynamically, though right now it's copied as a constant.  But I didn't see a pattern elsewhere for non-constant options or passing in option updates, so I'm assuming you plan on a refactoring to support that across all components in the future.

Reply:
If I understand the latest build output, it's complaining that it doesn't like "acn" anywhere because it thinks it's a misspelling?  :) Assume you will add that to the whitelist, or do I need to rebase to pick up that config change somehow?

Reply:
Thanks @brucelowekamp . Yeah it's more just if we can avoid creating this issue in the first place, it would make future updates a lot simpler if/when someone gets a chance to add that feature. We've got fake RDM control of some stuff like the SPI pixels, so you can configure them from our GUI or over Art-Net etc.

Reply:
> If I understand the latest build output, it's complaining that it doesn't like "acn" anywhere because it thinks it's a misspelling? :) Assume you will add that to the whitelist, or do I need to rebase to pick up that config change somehow?

Indeed that is the issue. I've pushed a change to fix it somewhere but I'm damned if I can find it. We just need to add a -L acn to the codespell command (although it's probably worth making it a variable as it's repeated twice currently):
https://github.com/codespell-project/codespell/blob/master/README.rst#usage

Reply:
@peternewman I pushed a fix that moves the ignore words to a separate file, then renamed the ignore lines file for clarity.  I thought it would be cleaner to keep that out of the build script.